lib/galaxy/jobs/runners/kubernetes.py

```diff
@@ -139,7 +139,10 @@ def queue_job(self, job_wrapper):
 
         job = Job(self._pykube_api, k8s_job_obj)
         job.create()
-        job_id = job.metadata['name']
+        if not job.name:
+            log.exception(f"failure creating k8s job: {job.obj}")
```

Comment:
I think you want to call fail with a user friendly job string to end the job evaluation based on the code earlier in this method:

```
job_wrapper.fail("Failed to create Kubernetes job properly.")
```

Maybe I'm wrong though?

Also you should properly tell the admin why we're failing here on a technical level also - even if we don't understand it exactly. I'd recommend:

```
log.exception(f"failure properly creating k8s job - empty name encountered: [{job.obj}]")
```

Reply:
I was mostly just copying the earlier error check.
I'll change up the text here and add the wrapper failure. Should the job_wrapper be failed at the other error check too?

Reply:
That was a dumb question, it is already there...